DNN Platform/DotNetNuke.Web.Client.ResourceManager/Models/ResourceBase.cs

```diff
@@ -0,0 +1,250 @@
+        /// <param name="htmlString">The HTML string builder to append to.</param>
+        protected void RenderIntegrity(StringBuilder htmlString)
+        {
+            if (!string.IsNullOrEmpty(this.Integrity))
+            {
+                htmlString.Append($" integrity=\"{this.Integrity}\"");
+            }
+        }
+
+        /// <summary>
+        /// Renders the referrer policy attribute.
+        /// </summary>
+        /// <param name="htmlString">The HTML string builder to append to.</param>
+        protected void RenderReferrerPolicy(StringBuilder htmlString)
+        {
+            if (this.ReferrerPolicy != ReferrerPolicy.None)
+            {
+                switch (this.ReferrerPolicy)
+                {
+                    case ReferrerPolicy.NoReferrer:
+                        htmlString.Append(" referrerpolicy=\"no-referrer\"");
+                        break;
+                    case ReferrerPolicy.NoReferrerWhenDowngrade:
+                        htmlString.Append(" referrerpolicy=\"no-referrer-when-downgrade\"");
+                        break;
+                    case ReferrerPolicy.Origin:
+                        htmlString.Append(" referrerpolicy=\"origin\"");
+                        break;
+                    case ReferrerPolicy.OriginWhenCrossOrigin:
+                        htmlString.Append(" referrerpolicy=\"origin-when-cross-origin\"");
+                        break;
+                    case ReferrerPolicy.SameOrigin:
+                        htmlString.Append(" referrerpolicy=\"same-origin\"");
+                        break;
+                    case ReferrerPolicy.StrictOrigin:
+                        htmlString.Append(" referrerpolicy=\"strict-origin\"");
+                        break;
+                    case ReferrerPolicy.StrictOriginWhenCrossOrigin:
+                        htmlString.Append(" referrerpolicy=\"strict-origin-when-cross-origin\"");
+                        break;
+                    case ReferrerPolicy.UnsafeUrl:
+                        htmlString.Append(" referrerpolicy=\"unsafe-url\"");
+                        break;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Renders the mimetype attribute.
+        /// </summary>
+        /// <param name="htmlString">The HTML string builder to append to.</param>
+        protected void RenderType(StringBuilder htmlString)
+        {
+            if (!string.IsNullOrEmpty(this.Type))
+            {
+                htmlString.Append($" type=\"{this.Type}\"");
+            }
+        }
+
+        /// <summary>
+        /// Renders the attributes.
+        /// </summary>
+        /// <param name="htmlString">The HTML string builder to append to.</param>
+        protected void RenderAttributes(StringBuilder htmlString)
+        {
+            foreach (var attribute in this.Attributes)
+            {
+                htmlString.Append($" {attribute.Key}=\"{attribute.Value}\"");
```

Comment:
We need to encode these attribute values
```suggestion
                htmlString.Append($" {attribute.Key}=\"{WebUtility.HtmlEncode(attribute.Value)}\"");
```